libmscore/edit.cpp

```diff
@@ -4537,10 +4538,10 @@ void Score::undoAddElement(Element* element)
             int staffIdx = staff->idx();
 
             QList<int> tr;
-            if ((strack & ~3) != staffIdx) // linked staff ?
-                  tr.append(staffIdx * VOICES + (strack % VOICES));
-            else if (staff->score()->excerpt() && strack > -1)
+            if (staff->score()->excerpt() && strack > -1)
                   tr = staff->score()->excerpt()->tracks().values(strack);
+            else if ((strack & ~3) != staffIdx) // linked staff ?
```

Comment:
I saw this test at several places but I think this is not correct to find linked staves. In PR #6049 I used another way to find linked parts and linked staves. This PR isn't merged in master so please have a look in <code>undoAddCR</code>.

Reply:
Thanks! I am now using your code, although it should probably be refactored in order to avoid code duplication.

Reply:
I'm working on a kind of "API" regarding linked staves and parts so we don't have to reinvent the same square wheel again :smiley: